From 1a61546e17af5cdcd7e2e06bb7c1f0baa2bb3685 Mon Sep 17 00:00:00 2001
From: Moudy Ho <moudy.ho@mediatek.com>
Date: Fri, 12 Sep 2025 10:40:15 +0800
Subject: [PATCH 4/9] GENIO: dt-bindings: media: mediatek: mdp3: Add support
 for MT8189 RDMA

Add a compatible for MediaTek MT8189 RDMA, which supports only a
subset of the MDP3 components of its similar MT8195 counterpart.

Change-Id: Ifb077c97479cfddb31c19f0e071bb7beb9a8952a
Signed-off-by: Moudy Ho <moudy.ho@mediatek.com>
---
 .../bindings/media/mediatek,mdp3-rdma.yaml     | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/Documentation/devicetree/bindings/media/mediatek,mdp3-rdma.yaml b/Documentation/devicetree/bindings/media/mediatek,mdp3-rdma.yaml
index 18603f6c5e06..23fc287ce080 100644
--- a/Documentation/devicetree/bindings/media/mediatek,mdp3-rdma.yaml
+++ b/Documentation/devicetree/bindings/media/mediatek,mdp3-rdma.yaml
@@ -24,6 +24,7 @@ properties:
       - enum:
           - mediatek,mt8183-mdp3-rdma
           - mediatek,mt8188-mdp3-rdma
+          - mediatek,mt8189-mdp3-rdma
           - mediatek,mt8195-mdp3-rdma
           - mediatek,mt8195-vdo1-rdma
       - items:
@@ -115,6 +116,23 @@ allOf:
         - mboxes
         - mediatek,gce-events
 
+  - if:
+      properties:
+        compatible:
+          contains:
+            const: mediatek,mt8189-mdp3-rdma
+
+    then:
+      properties:
+        clocks:
+          maxItems: 1
+
+        mboxes:
+          minItems: 3
+
+      required:
+        - mediatek,gce-events
+
   - if:
       properties:
         compatible:
-- 
2.45.2

